pom.xml

```diff
@@ -899,10 +920,30 @@
                     <artifactId>java17</artifactId>
                     <version>1.0</version>
                   </signature>
+                  <ignores>
+                    <ignore>org.apache.hadoop.conf.Configuration</ignore>
+                    <ignore>org.apache.hadoop.fs.FileSystem</ignore>
+                  </ignores>
                 </checkSignatureRule>
               </rules>
             </configuration>
           </execution>
+          <execution>
+            <id>enforce-hadoop-profile</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <requireProperty>
+                  <property>hadoop.profile</property>
+                  <regex>(2|3)</regex>
+                  <regexMessage>You should specify the Hadoop profile by major Hadoop generation, i.e. 2 or 3, not by a version number.
+    Use hadoop.version to use a particular Hadoop version within that generation.</regexMessage>
+                </requireProperty>
+              </rules>
```

Comment:
If 3 executions of the enforcer plugin is too slow, we could consider rolling this rule into one of the other executions... probably the basic one. The only reason we have the second execution for the animal sniffer, is because it needs to run after compilation. This rule could run at the beginning with the other basic rules. It's fine separate, too. This is just a thought, in case it's slow.